miso-web/src/main/webapp/scripts/project_ajax.js

```diff
@@ -47,14 +53,28 @@ var Project = (function () {
 
     addContact: function (addContact) {
       var contacts = Project.getContacts();
-      contacts.push(addContact);
-      Project.setContacts(contacts);
+      addContact["projectId"] = projectId;
+      if (
+        contacts.find(function (contact) {
```

Comment:
Added this check because `setTableData`'s fake IDs would have weird behaviour in the following instance:

A duplicate contact+role is added, then the user attempts to save, resulting in the error message describing that there is a duplicate contact. When you try to select any item in the list after, even if you check the box, it does not select anything.

By adding this, it prevents duplication, and thus mitigates the potential of this error. The duplication check in `DefaultProjectService` should still be kept in the odd chance that somehow this check fails, as we don't want to cause an error in the database and corrupt data.